scala/private/phases/phase_collect_jars.bzl

```diff
@@ -46,41 +41,32 @@ def phase_collect_jars_junit_test(ctx, p):
     )
     return _phase_collect_jars_default(ctx, p, args)
 
-def phase_collect_jars_library_for_plugin_bootstrapping(ctx, p):
-    args = struct(
-        unused_dependency_checker_mode = "off",
-    )
-    return _phase_collect_jars_default(ctx, p, args)
-
 def phase_collect_jars_common(ctx, p):
     return _phase_collect_jars_default(ctx, p)
 
 def _phase_collect_jars_default(ctx, p, _args = struct()):
     return _phase_collect_jars(
-        ctx,
-        _args.base_classpath if hasattr(_args, "base_classpath") else p.scalac_provider.default_classpath,
-        _args.extra_deps if hasattr(_args, "extra_deps") else [],
-        _args.extra_runtime_deps if hasattr(_args, "extra_runtime_deps") else [],
-        _args.unused_dependency_checker_mode if hasattr(_args, "unused_dependency_checker_mode") else p.unused_deps_checker,
+        ctx = ctx,
+        p = p,
+        base_classpath = _args.base_classpath if hasattr(_args, "base_classpath") else p.scalac_provider.default_classpath,
+        extra_deps = _args.extra_deps if hasattr(_args, "extra_deps") else [],
+        extra_runtime_deps = _args.extra_runtime_deps if hasattr(_args, "extra_runtime_deps") else [],
     )
 
 # Extract very common code out from dependency analysis into single place
 # automatically adds dependency on scala-library and scala-reflect
 # collects jars from deps, runtime jars from runtime_deps, and
 def _phase_collect_jars(
         ctx,
+        p,
```

Comment:
wdyt about passing in `p.dependency`? `p` is a very big struct to pass and widens the method (as a reader I'm now more alert)

Reply:
Sounds good